### DISCUSSION TITLE
docker/ghcr.io: unable to retrieve latest tag / incomplete pagination?

### DISCUSSION THREAD

**Discussion by maxbrunet** [renovatebot/renovate#24638] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I am looking at a missing update for the `ghcr.io/parca-dev/parca` docker image, its current version is `v0.18.0`, and the latest is [`v0.19.0`](https://github.com/orgs/parca-dev/packages/container/parca/130196251?tag=v0.19.0), but Renovate does not find it.

I see these HTTP statistics:

```jsonc
    "https://ghcr.io/token (GET,200)": 1,
    "https://ghcr.io/v2/ (GET,401)": 1,
    "https://ghcr.io/v2/parca-dev/parca/tags/list (GET,200)": 4,
    "https://ghcr.io/v2/parca-dev/parca/tags/list (GET,401)": 1
```

Since this repository has 4254 tags and `ghcr.io` returns 1000 tags per request at most, I would expect to see 5 `/tags/list (GET,200)`, but Renovate only makes 4:

```console
$ regctl tag ls ghcr.io/parca-dev/parca >/dev/null | wc -l
4254
```

From some manual tests, `v0.18.0` would be at index `3373` (page 4), while `v0.19.0` would be at `4208` (page 5).

Is `RENOVATE_X_DOCKER_MAX_PAGES` set to 4 in the hosted app?

https://github.com/renovatebot/renovate/blob/8d1f9d4b9838b96d510831eb95e776fc6f675ee6/lib/modules/datasource/docker/index.ts#L644-L646

Minimal reproduction: https://github.com/maxbrunet/renovate-docker-ghcr.io-pagination

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "dockerfile": [
      {
        "deps": [
          {
            "depName": "ghcr.io/parca-dev/parca",
            "currentValue": "v0.18.0",
            "replaceString": "ghcr.io/parca-dev/parca:v0.18.0",
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "datasource": "docker",
            "depType": "final",
            "updates": [],
            "packageName": "ghcr.io/parca-dev/parca",
            "versioning": "docker",
            "warnings": [],
            "sourceUrl": "https://github.com/parca-dev/parca",
            "registryUrl": "https://ghcr.io",
            "currentVersion": "v0.18.0",
            "fixedVersion": "v0.18.0"
          }
        ],
        "packageFile": "Dockerfile"
      }
    ]
  }
}
```

</details>

I do not have full logs from the reproduction where Renovate fetches the tags, because of the caching. And logs from actual repositories are very noisy.

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Does GHCR sort tags from oldest to newest? I think we may have pages limitation in the app with the assumption it would be the 4000 newest tags, not oldest. Cc @nabeelsaabna

**Reply by viceice** [reply]
max pages is 20 by default

https://github.com/renovatebot/renovate/blob/8d1f9d4b9838b96d510831eb95e776fc6f675ee6/lib/modules/datasource/docker/index.ts#L644-L646

**Reply by maxbrunet** [reply]
> Does GHCR sort tags from oldest to newest?

Yes, tags are in chronological order.

I tested with self-hosted and there are no issues, all 5 pages are retrieved.

**Reply by rarkins** [reply]
Yes, the hosted renovate app limits to 4 pages currently. This was mainly to reduce load/API calls on Docker Hub. We can increase to 5 today

Cc @nabeelsaabna

**Reply by viceice** [reply]
@rarkins I think you can even do a higher page count in app, as for docker hub there is a different code path which uses the hub API instead the normal docker api.

**Reply by nabeelsaabna** [reply, marked as answer]
bumped to 7

**Reply by maxbrunet** [reply]
Renovate finds the update now, thank you!

I have opened an issue on the Parca repository to look into clean up old tags, so we do not hit the new limit.
